File: app/src/main/java/com/ed/redditapp/lib/api/standard/StandardSearchItemSubreddit.java
package com.ed.redditapp.lib.api.standard;

import com.ed.redditapp.lib.api.SearchItemSubreddit;

import org.json.JSONException;
import org.json.JSONObject;

import lombok.Getter;

@Getter
public class StandardSearchItemSubreddit implements SearchItemSubreddit {
    private final String name;
    private final int subsCount;

    public StandardSearchItemSubreddit(JSONObject json) throws JSONException {
        name = json.getString("display_name");
        subsCount = json.getInt("subscribers");
    }
}

File: app/src/main/java/com/ed/redditapp/lib/api/standard/StandardSearchItemSubreddit.kt
package com.ed.redditapp.lib.api.standard

import com.ed.redditapp.lib.api.SearchItemSubreddit
import org.json.JSONObject

class StandardSearchItemSubreddit(json: JSONObject) : SearchItemSubreddit {
    override val name = json.getString("display_name")
    override val subsCount = json.getInt("subscribers")
}